Segment-Optimizely-X/Classes/SEGOptimizelyXIntegration.m

```diff
@@ -121,21 +121,24 @@ - (void)experimentDidGetViewed:(NSNotification *)notification
 
 - (void)enqueueAction:(SEGTrackPayload *)payload
 {
-    SEGLog(@"%@ Optimizely not initialized. Enqueueing action: %@", self, payload);
-    @try {
-        //TO DO: Should we set a condition to stop queuing events?
-        // Should set timer? Android has 5 minutes before stopping
-        if (self.queue.count > 1000) {
-            // Remove the oldest element.
-            [self.queue removeObjectAtIndex:0];
-            SEGLog(@"%@ removeObjectAtIndex: 0", self.queue);
+    @synchronized(payload)
```

Comment:
why do you have this?

Reply:
flushQueue is called on the main thread. enqueueAction is called by track on a background thread. Hence the queue is accessed on multiple threads, and needs synchronization.

Reply:
so my suggestion to ladan was to always dispatch to the queue to execute.

Reply:
Yes we can either keep it thread confined, or use locks. Either one is fine.